### PR TITLE
[Core] Support relative path in Polling Strategy 

### PR DESCRIPTION
**Pull request by mssfang**
Based on the request of https://github.com/Azure/azure-sdk-for-java/issues/29375

Added the support for relative path in Polling Strategy.

### CODE REVIEW

**Review by weidongxu-microsoft — COMMENTED** [review]
Overall LGTM.

Wait for review from core experts.

File: sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java
Comment (haolingdong-msft) [inline sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java:130]: Do we need to override getResult() to make get final result url absolute path as well?

Comment (weidongxu-microsoft) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java:130]: `response.getRequest().getUrl()` is from a working request and must be absolute.

File: sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java
Comment (weidongxu-microsoft) [inline sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: the case in Haoling's comment is from deviceupdate that the final resource URL is also relative

e.g.
```
"resourceLocation": "/deviceUpdate/sdkinstance/updates/providers/fabrikam/names/vacuum/versions/2022.609.510.21?api-version=2022-07-01-preview"
```

Comment (weidongxu-microsoft) [reply sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: @mssfang Any update, as code freeze is closing?

Comment (mssfang) [reply sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: Sorry for the late reply. Do you mean resourceLocation could be either an absolute or relative path?

Comment (weidongxu-microsoft) [reply sdk/core/azure-core/src/test/java/com/azure/core/util/polling/PollingStrategyTests.java:268]: Yes, in the case of deviceupdate backend, we found it is also relative (the ref on earlier comment is the real one we get).

File: sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java
Comment (mssfang) [reply sdk/core/azure-core/src/main/java/com/azure/core/util/polling/OperationResourcePollingStrategy.java:130]: updated

**Review by weidongxu-microsoft — APPROVED** [review]
LGTM.

@alzimmermsft @srnagar for review from core.

**Review by alzimmermsft — APPROVED** [review]

**Review by srnagar — APPROVED** [review]

**Pull request merged**
